casper/src/main/scala/coop/rchain/casper/blocks/proposer/Proposer.scala

```diff
@@ -84,18 +84,23 @@ class Proposer[F[_]: Concurrent: Log: Span](
   private def checkProposeConstraints(
       genesis: BlockMessage,
       s: CasperSnapshot[F]
-  ): F[CheckProposeConstraintsResult] = {
-    val s1   = Stream[F, CheckProposeConstraintsResult](checkActiveValidator(s, validator))
-    val s2   = Stream.eval[F, CheckProposeConstraintsResult](checkEnoughBaseStake(genesis, s))
-    val s3   = Stream.eval[F, CheckProposeConstraintsResult](checkFinalizedHeight(genesis, s))
-    val work = Stream(s1, s2, s3)
-    work
-      .parJoin(3)
-      .compile
-      .toList
-      // pick some result that is not Success, or return Success
-      .map(_.find(_ != CheckProposeConstraintsSuccess).getOrElse(CheckProposeConstraintsSuccess))
-  }
+  ): F[CheckProposeConstraintsResult] =
+    checkActiveValidator(s, validator) match {
```

Comment:
As sync constraints depend on last message - for not bonded validators there is no such message. So first we check bonding and then check constraint, that might rise an error if no latest message for bonded validator found.